### PR TITLE
conncache: make hashkey avoid malloc

### PR DESCRIPTION
**Pull request by bagder**
... to make it much faster. Idea developed with primepie on IRC.

### CODE REVIEW

File: lib/conncache.c [hunk 1]
```diff
@@ -141,7 +142,11 @@ static char *hashkey(struct connectdata *conn)
   else
     hostname = conn->host.name;
 
-  return aprintf("%s:%ld", hostname, conn->port);
+  DEBUGASSERT(len > 32);
+
+  /* put the number first so that the hostname gets cut off if too long */
+  snprintf(buf, len, "%ld%s", conn->port, hostname);
```
Comment (MarcelRaad) [inline lib/conncache.c]: Isn't snprintf C99? I believe MSVC versions prior to 2015 only have a _snprintf that doesn't append a '\0' if the string gets truncated.

Comment (bagder) [inline lib/conncache.c]: That's one of the reasons why have our own set of *printf() implementations and snprintf comes from here: https://github.com/curl/curl/blob/master/lib/curlx.h#L105

If you check the code closer you'll see that we already use snprintf() in numerous places thanks to this.

Comment (MarcelRaad) [inline lib/conncache.c]: Ah, great! I forgot about that.

File: lib/conncache.c [hunk 2]
```diff
@@ -128,7 +128,8 @@ void Curl_conncache_destroy(struct conncache *connc)
 }
 
 /* returns an allocated key to find a bundle for this connection */
```
Comment (bagder) [inline lib/conncache.c]: I'll fix the comment too.

**Comment by bagder** [comment]
thanks for the review!

**Pull request closed without merging**